pkg/generate/app/cmd/template.go

```diff
@@ -38,11 +38,7 @@ func TransformTemplate(tpl *templateapi.Template, client client.TemplateConfigsN
 	}
 
 	// ensure the template objects are decoded
-	// TODO: in the future, this should be more automatic
-	// TODO: this should use the UniversalDecoder() once we deprecate legacy group. Until
-	// then we have to maintain the backward compatibility with old servers so this will
-	// force to decode the list as legacy/core API.
-	if errs := runtime.DecodeList(result.Objects, kapi.Codecs.LegacyCodec(latest.Version)); len(errs) > 0 {
+	if errs := runtime.DecodeList(result.Objects, kapi.Codecs.UniversalDecoder(), unstructured.UnstructuredJSONScheme); len(errs) > 0 {
```

Comment:
@deads2k i could have sworn we went through this before, are we supposed to have to have changed this?

Also the comments that are being removed worry me a little, is this going to break new-app for using legacy api objects?

Reply:
(or break compatibility w/ old servers)

Reply:
> (or break compatibility w/ old servers)

On decoding?  I don't think so.  This is probably safer if new-app can tolerate it.  I don't know enough about new-app to reason about its safety. Do you ever type-cast an object?

Reply:
I think this type switch is the closest we come:
https://github.com/openshift/origin/blob/master/pkg/util/labels.go#L42

Reply:
but these objects will get posted back to the server.  so if we've converted them from legacy to group resources, we're going to fail when we go to create them on the legacy server, right?

Reply:
> I think this type switch is the closest we come:
> https://github.com/openshift/origin/blob/master/pkg/util/labels.go#L42

Well, this change will make that fail.



> but these objects will get posted back to the server. so if we've converted them from legacy to group resources, we're going to fail when we go to create them on the legacy server, right?

Unstructured does't do any conversion.

Reply:
> I think this type switch is the closest we come:
> https://github.com/openshift/origin/blob/master/pkg/util/labels.go#L42
> Well, this change will make that fail.

I think.  Maybe only on Tuesdays.  New-app is a pain :(

Reply:
I would argue our api/conversions are a pain. :)

Reply:
(we also use AddObjectLabels which does a similar type switch.  So we need to understand if this change is going to result in those annotations+labels not getting added when new-app is run against an older server)

Reply:
We might have to run through both codepaths....  use the legacy decoder, if we hit a resource we don't understand, go back and use the universaldecoder instead.

Reply:
@deads2k what level do we have to go back to to only have the legacy api? i want to test this.

Reply:
> @deads2k what level do we have to go back to to only have the legacy api? i want to test this.

3.5